daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -184,6 +185,38 @@ func (l *logStream) Close() error {
 
 // create creates a log stream for the instance of the awslogs logging driver
 func (l *logStream) create() error {
+
+	// Create log group first
+	{
+		input := &cloudwatchlogs.CreateLogGroupInput{
+			LogGroupName: aws.String(l.logGroupName),
+		}
+
+		_, err := l.client.CreateLogGroup(input)
+
+		if err != nil {
+			if awsErr, ok := err.(awserr.Error); ok {
+				fields := logrus.Fields{
+					"errorCode":     awsErr.Code(),
+					"message":       awsErr.Message(),
+					"origError":     awsErr.OrigErr(),
+					"logGroupName":  l.logGroupName,
+					"logStreamName": l.logStreamName,
+				}
+				if awsErr.Code() == resourceAlreadyExistsCode {
```

Comment:
You'll also want to explicitly handle permission problems here, in case the IAM policy is such that `logs:CreateLogGroup` is not allowed but `logs:CreateLogStream` and `logs:PutLogEvents` are.